.github/workflows/ci.yaml

```diff
@@ -1,73 +1,58 @@
 name: ci
 on:
   push:
-    branches: [main]
-  pull_request:
-    branches: [main]
-  schedule:
-    - cron: '15 22 * * *'
-  workflow_dispatch: {} # support manual runs
-# Prevent writing to the repository using the CI token.
-# Ref: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions
-permissions: read-all
+  workflow_dispatch:
 env:
-  SERVICE: connect-demo
-  REGISTRY: connect-demo
-  REGION: us-central1
+  MAKEFLAGS: "-j 8"
+  PRIVATE_REGISTRY_HOST: us.gcr.io
+  PRIVATE_REGISTRY_URL: us.gcr.io/buf-private-gcr-1
 jobs:
-  ci:
+  test:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout Code
+      - name: checkout
         uses: actions/checkout@v3
-        with:
-          fetch-depth: 1
-      - name: Install Go
+      - name: setup-go
+        if: success()
         uses: actions/setup-go@v3
         with:
           go-version: 1.18.x
-      - name: Cache
-        uses: actions/cache@v3
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-connect-demo-ci-${{ hashFiles('**/go.sum') }}
-          restore-keys: ${{ runner.os }}-connect-demo-ci-
-      - name: Test
+      - name: make-test
+        if: success()
         run: make test && make checkgenerate
-      - name: Lint
+  lint:
```

Comment:
Why separate this from running tests? I'd expect it to duplicate quite a bit of work, since actually running tests and linters is quite fast.

Reply:
My main motivation is to match our internal configurations. We do lose some duplicate work but we also gain parallelism. I think engineering time is more important, but no strong feelings here - the difference should be small. Combined them in one workflow.